src/overlays/actors/ovl_En_Hgo/z_en_hgo.c

```diff
@@ -22,54 +52,344 @@ const ActorInit En_Hgo_InitVars = {
     (ActorFunc)EnHgo_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BD0F48 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ActorAnimationEntry sAnimations[] = {
+    { &D_0600B644, 1.0f, 0.0f, 0.0f, 0, -4.0f }, { &D_06013684, 1.0f, 0.0f, 0.0f, 0, 0.0f },
+    { &D_060152EC, 1.0f, 0.0f, 0.0f, 2, 0.0f },  { &D_06015C70, 1.0f, 0.0f, 0.0f, 0, 0.0f },
+    { &D_060165F0, 1.0f, 0.0f, 0.0f, 0, 0.0f },  { &D_06014220, 1.0f, 0.0f, 0.0f, 2, 0.0f },
+    { &D_06014A9C, 1.0f, 0.0f, 0.0f, 0, 0.0f },
+};
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 18, 46, 0, { 0, 0, 0 } },
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit2 D_80BD0F74 = { 0, 0, 0, 0, MASS_IMMOVABLE };
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+// Eye Textures
+static UNK_TYPE4* D_80BD0F80[] = {
+    &D_06011138,
+    &D_06011938,
+    &D_06012138,
+};
+
+void EnHgo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHgo* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&thisx->shape, 0.0f, func_800B3FC0, 36.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06012A58, &D_0600B644, this->limbDrawTbl,
+                       this->transitionDrawTbl, 19);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&thisx->colChkInfo, NULL, &sColChkInfoInit);
+    thisx->targetMode = 6;
+    this->unk_30C = 0;
+    this->unk_30E = 0;
+    this->unk_314 = 0;
+    this->unk_310 = 0;
+    this->unk_312 = 0;
+    if (gSaveContext.weekEventReg[75] & 0x20 || gSaveContext.weekEventReg[52] & 0x20) {
```

Comment:
We usually put parenthesis around "bitwise and" operations
```suggestion
    if ((gSaveContext.weekEventReg[75] & 0x20) || (gSaveContext.weekEventReg[52] & 0x20)) {
```